### ISSUE TITLE
Submitty Database Migration Error:  Database does not exist for master

### ISSUE THREAD

**Issue by erelsgl** [Submitty/Submitty#6210]
**Describe the bug**
During installation, after I answer all configuration questions, I get the following error:

`
Beginning Install Submitty Script
bash ${SUBMITTY_INSTALL_DIR}/.setup/INSTALL_SUBMITTY.sh clean skip_web_restart
CLONE OR UPDATE AnalysisTools...
    current version matches or exceeds minimum version v.18.06.00
CLONE OR UPDATE Lichen...
    current version matches or exceeds minimum version v20.05.00
CLONE OR UPDATE RainbowGrades...
    current version matches or exceeds minimum version v21.01.01
CLONE OR UPDATE Tutorial...
    current version matches or exceeds minimum version v20.10.01
CLONE OR UPDATE SysadminTools...
    current version matches or exceeds minimum version v20.01.00
Checking for system and database migrations
Submitty Database Migration Error:  Database does not exist for master
err_message
`

**Expected behavior**
I expected the install to complete successfully.

**To Reproduce**
1. I just run bash ./.setup/install_system.sh and answer the questions with mostly the default values.
2. See error

**Configuration**
 - OS: Ubuntu
 - Browser: irrelevant

**Comment by erelsgl** [comment]
Note: `systemctl status postgresql` shows:

```
● postgresql.service - PostgreSQL RDBMS
   Loaded: loaded (/lib/systemd/system/postgresql.service; enabled; vendor preset: enabled)
   Active: active (exited) since Wed 2021-02-17 21:23:09 UTC; 2min 52s ago
  Process: 17225 ExecStart=/bin/true (code=exited, status=0/SUCCESS)
 Main PID: 17225 (code=exited, status=0/SUCCESS)

Feb 17 21:23:09 badkan2.com systemd[1]: Starting PostgreSQL RDBMS...
Feb 17 21:23:09 badkan2.com systemd[1]: Started PostgreSQL RDBMS.
```

**Comment by MasterOdin** [comment]
For the postgres server, what databases exist in it?

**Comment by erelsgl** [comment]
```
postgres=# \l
                                 List of databases
   Name    |      Owner      | Encoding | Collate |  Ctype  |   Access privileges
-----------+-----------------+----------+---------+---------+-----------------------
 postgres  | postgres        | UTF8     | C.UTF-8 | C.UTF-8 |
 submitty  | submitty_dbuser | UTF8     | C.UTF-8 | C.UTF-8 |
 template0 | postgres        | UTF8     | C.UTF-8 | C.UTF-8 | =c/postgres          +
           |                 |          |         |         | postgres=CTc/postgres
 template1 | postgres        | UTF8     | C.UTF-8 | C.UTF-8 | =c/postgres          +
           |                 |          |         |         | postgres=CTc/postgres
(4 rows)
```

**Comment by hdrodz97** [comment]
I'm getting the impression that there's something preventing Submitty from talking to Postgres.

It seems like the source of the error is [here](https://github.com/Submitty/Submitty/blob/master/migration/migrator/main.py#L208-L214) (snippet reproduced below):
```python
try:
    database = db.Database(loop_args.config.database, environment)
except OperationalError:
    raise SystemExit(
        'Submitty Database Migration Error:  '
        'Database does not exist for {}'.format(environment)
    )
```
Is there any reason we're not at least printing the exception message? Seems like the "database does not exist" message is a false lead since the migrator's trying to connect to the `submitty` database, which does exist.

**Comment by shailpatels** [comment]
@hdrodz97 Yeah its an annoying quirk since an OperationalError could be thrown from just unable to connect, I think printing the exception string would be helpful.

@erelsgl Do you remember what configuration answers you used when installing submitty initially? There might be a bug from a certain input causing the database to silently fail.

**Comment by MasterOdin** [comment]
To give more context on this, you can double check that the values in `/usr/local/submitty/config/database.json` match what you expect, and can be used to connect to the database using something like `psql`.

**Comment by erelsgl** [comment]
Here is the content of database.json:

```
{
  "authentication_method": "DatabaseAuthentication",
  "database_host": "localhost",
  "database_port": 5432,
  "database_user": "submitty_dbuser",
  "database_password": "",
  "debugging_enabled": false
}

```

I can connect to psql in the following way:

```
# sudo -u postgres psql
psql (10.15 (Ubuntu 10.15-0ubuntu0.18.04.1))
Type "help" for help.

postgres=# \l
                                 List of databases
   Name    |      Owner      | Encoding | Collate |  Ctype  |   Access privileges
-----------+-----------------+----------+---------+---------+-----------------------
 postgres  | postgres        | UTF8     | C.UTF-8 | C.UTF-8 |
 submitty  | submitty_dbuser | UTF8     | C.UTF-8 | C.UTF-8 |
 template0 | postgres        | UTF8     | C.UTF-8 | C.UTF-8 | =c/postgres          +
           |                 |          |         |         | postgres=CTc/postgres
 template1 | postgres        | UTF8     | C.UTF-8 | C.UTF-8 | =c/postgres          +
           |                 |          |         |         | postgres=CTc/postgres
(4 rows)
```

**Comment by erelsgl** [comment]
UPDATE: I installed submitty on a fresh Ubuntu 18.04 server, and I get the same error.

**Comment by MasterOdin** [comment]
How did you create the `submitty_dbuser` DB user? Did you give it a password within the database? Does `pg_hba.conf` allow for local connection using `md5` (as opposed to just `peer`)?

**Comment by erelsgl** [comment]
I did not create it - I just ran the installation script. According to the [instructions](https://submitty.org/sysadmin/installation),

**Comment by jingmingcn** [comment]
@erelsgl 
I got the same issue just like your description and is there any update about the issue or is there any other trick that could be used to work around. Thanks.

**Comment by MasterOdin** [comment]
The next thing to test would be that doing:

```
psql -U submitty_dbuser -d submitty -W
```

and then inputting the password set in `/usr/local/submitty/config/database.json` works. I plan to also land a patch soon to make this error message more informative to make debugging a bit easier.
